Data/TLS/PThread/Internal.hs

```diff
@@ -92,6 +100,13 @@ getTLS TLS{key,mknew,allCopies} = do
     return a
    else
     deRefStablePtr p
+ where
+  checkBoundness = when rtsSupportsBoundThreads $ do
+    b <- isCurrentThreadBound
+    when (not b) $ do
+      let msg = "thread-local-storage: PThread.getTLS used from an unbound thread."
+      hPutStrLn stderr msg
+      error msg
```

Comment:
Do we need to both print the message to `stderr` and call `error`? Alternatively, couldn't we just call `fail`?